index.js

```diff
@@ -470,7 +435,7 @@ module.exports = (robot, _, Settings = require('./lib/settings')) => {
     }
     const suborg = getChangedConfigName(new Glob(".github/suborgs/*.yml"), files, context.repo().owner)
     if (suborg) {
-      return syncAllSettings(true, context, suborg, pull_request.head.ref )
+      return syncSubOrgSettings(true, context, suborg, context.repo(), pull_request.head.ref )
```

Comment:
shouldn't the 
```
return syncAllSettings(true, context, context.repo(), pull_request.head.ref )
```

be the first thing to be done if there are settings? I mean if there is a `settings.yml` we are going to to a synch all anyway, not need to evaluate suborgs (and also handle the case if someone pushes a suborg and a settings in the same push)

something like
```
    if (settingsModified) {
      robot.log.debug(`Changes in '${Settings.FILE_NAME}' detected, doing a full synch...`)
      return syncAllSettings(false, context)
    }
```

Reply:
I'm talking about the end of this block (but can't put a comment on something that hasn't changed).

But this is getting associated to the change from ` syncAllSettings`

This is confusing :)

Reply:
Yes, I agree. Moved that logic up.

Reply:
@tspascoal I made changes to safe-settings where it will handle multiple files in the commit:
If it is settings.yml it will do syncall, otherwise it will do sync for the repos and  suborgs iteratively. 

I also made changes so changes will be done only if there are actual changes based on deepcompare. 

I did some scale testing with 1000 repos, with those modifications, I am seeing the following performance improvements:
1. Trigger sync all with changes to settings.yml but there are no changes in the settings: 2 minutes to fetch the current state for 1000 repos and compare the settings to determine there are no changes.
2. Trigger sync all with changes to settings.yml to add repo topics to all 1000 repos:  15 minutes
3. Changes to settings.yml to add branch protections to all the 1000 repos: 20 minutes
4. Run the PR check and perform validations for 1000 repos based on a change to settings.yml - 5 minutes

Probot automatically handles rate and abuse limiting. These numbers are including the delays to handle abuse limits and rate limits.  

I had to make changes to the way PR checks were reporting the status. Previously I was creating a detailed report which fails when completing the 1check` because the payload size was too large. So I changed it to report just the summary like:
<img width="886" alt="image" src="https://user-images.githubusercontent.com/57544838/163307103-20847831-6856-4405-bacd-0b92b59383c1.png">